src/auth/authenticator/api-key/authenticator.go

```diff
@@ -0,0 +1,88 @@
+package apikey
+
+import (
+	"encoding/base64"
+	"encoding/hex"
+	"hash"
+	"net/http"
+	"strings"
+
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/types"
+)
+
+const (
+	AuthMode    = "ApiKey"
+	BasicSchema = "Basic"
+)
+
+type Authenticator struct {
+	APIKeyFile map[string]*UserNameAndGroups
+	Hasher     *hash.Hash
+	B64Encoder *base64.Encoding
+}
+
+func NewAuthenticator(cfg *Config) (*Authenticator, error) {
+	if len(cfg.APIKeyFile) == 0 {
+		return nil, nil
+	}
+
+	auth := &Authenticator{APIKeyFile: cfg.APIKeyFile,
+		Hasher:     cfg.Hasher,
+		B64Encoder: cfg.B64Encoder,
+	}
+
+	return auth, nil
+}
+
+// Authenticate checks APIKEY hashes retrieve user Info
+// ? -> Username
+// ? -> Groups
+func (authenticator Authenticator) Authenticate(req *http.Request) (*types.UserInfo, error) {
+	// extract ApiKey
+	clientAPIKey, ok, err := extractAPIKey(req.Header.Get("Authorization"), authenticator.B64Encoder)
+	if err != nil {
+		// could not be decoded
+		return nil, errors.UnauthorizedError("apikey format error")
+	}
+	// respond nothing when none found
+	if !ok || clientAPIKey == "" {
```

Comment:
As far as I can see `ok == true && clientAPIKey == ""` cannot happen without an error calling `extractAPIKey()`

Reply:
Yes, right. I did simplify this bit